cmd/traffic_cache_tool/CacheTool.cc

```diff
@@ -521,6 +534,30 @@ Stripe::dir_freelist_length(int s)
   return free;
 }
 
+int
+Stripe::check_loop(int s)
+{
+  // look for loop in the segment
+  // rewrite the freelist if loop is present
```

Comment:
Isn't this just the detection part?

Reply:
yes it is. This command is just a note for myself.  I have the rewrite part in dir_init_segment.